packages/slice-machine/src/hooks/useRouteChange.tsx

```diff
@@ -0,0 +1,42 @@
+import { type NextRouter, useRouter } from "next/router";
+import {
+  type FC,
+  type PropsWithChildren,
+  createContext,
+  useContext,
+  useState,
+} from "react";
+
+type RouteChange = Readonly<{ source: Route; destination: Route }>;
+
+export type Route = Readonly<Pick<NextRouter, "asPath" | "query">>;
+
+const RouteChangeContext = createContext<RouteChange | undefined>(undefined);
+
+export const RouteChangeProvider: FC<PropsWithChildren> = (props) => (
+  <RouteChangeContext.Provider {...props} value={_useRouteChange()} />
```

Comment:
Is this a pattern we want to add in the code guideline? Starting a function by "_" when we want to keep it private?